src/game_state.ts

```diff
@@ -358,14 +356,19 @@ export class GameStateController {
             // @ts-ignore: There is hell and then there is writing types for external APIs
             const currentPhase = this.decodePhase(response.phase);
             if (currentPhase != this.state.storyletPhase) {
-                if (currentPhase == StoryletPhases.End) {
-                    // @ts-ignore: There is hell and then there is writing types for external APIs
-                    this.state.storyletId = response.endStorylet.event.id;
-                }
-
```

Comment:
If we remove this, then the following two cases that you introduced also need to update `this.state.storyletId`, as it is a separate variable. Or maybe introduce an accessor that will automatically pull that value from the `currentStorylet`?

Reply:
Took me a while to puzzle back together what happened here. My understanding is that the `storyletId` variable was redundant, because all accessors _except `matchers.ts`_, which is changed accordingly below, used `state.currentStorylet.id` instead.